src/Microsoft.AspNetCore.OData/Adapters/WebApiContext.cs

```diff
@@ -83,14 +84,39 @@ public string RouteName
         }
 
         /// <summary>
-        /// Gets or sets the parsed OData <see cref="SelectExpandClause"/> of the request.
+        /// Gets or sets the processed OData <see cref="SelectExpandClause"/> of the request. It will be different from QueryOptions.SelectExpand. SelectExpandClause as the $level is processed in this clause. 
         /// </summary>
-        public SelectExpandClause SelectExpandClause
+        public SelectExpandClause ProcessedSelectExpandClause
         {
             get { return this.innerFeature.SelectExpandClause; }
             set { this.innerFeature.SelectExpandClause = value; }
         }
 
+        /// <summary>
+        /// Gets or sets the parsed <see cref="ODataQueryOptions"/> of the request.
+        /// </summary>
+        public ODataQueryOptions QueryOptions
+        {
+            get
+            {
+                ODataFeature feature = this.innerFeature as ODataFeature;
+                if (feature != null)
+                {
+                    return feature.QueryOptions;
+                }
+
+                return null;
+            }
+            set
+            {
+                ODataFeature feature = this.innerFeature as ODataFeature;
+                if (feature != null)
```

Comment:
>feature != null [](start = 20, length = 15)

We should have a comment that explains what we are doing here -- we are only able to save the QueryOptions on the innerFeature if it is an ODataFeature, because IODataFeature doesn't support QueryOptions in 7.x.  We can fix this in 8.x. In the meantime, if this.innerFeature is not an ODataFeature, should we save QueryOptions in a private field, so if the user sets and then gets QueryOptions they get what they set, and not null?